src/Rings/mpoly-graded.jl

```diff
@@ -1496,6 +1501,157 @@ function homogenization(f::MPolyElem, S::MPolyRing_dec, pos::Int = 1)
   return finish(B)
 end
 
+function _homogenization(f::MPolyElem, S::MPolyRing_dec, start_pos::Int = 1)
+   len_gg  = ngens(S.D)
+      #= get exponent vectors and enlarge them for S =#
+   exps = Vector{Int}[]
+   for e in exponent_vectors(f)
+       for j in 1:len_gg
+           insert!(e, start_pos+j-1, 0)
+       end
+       push!(exps, e)
+   end
+
+   #= compute degrees in S =#
+   if is_z_graded(S) == true
+       degs    = Int[]
+       gens_gg = [Int(S.d[i][1])  for i in 1:ngens(S)]
+   else
+       degs    = Vector{Int}[]
+       gens_gg = [[Int(S.d[i][j]) for j in 1:ngens(S.D)] for i in 1:ngens(S)]
+   end
+
+   for e in exps
+       push!(degs, sum((e.*gens_gg)))
+   end
+   # get top degrees
+   tdeg  = vcat(maximum(hcat(degs...), dims=2)...)
+       #= finally generate homogeneous version of f in S, called F =#
+   F  = MPolyBuildCtx(S)
+
+   for (i, (c, e)) = enumerate(zip(coefficients(f), exps))
+       for j in 1:len_gg
+           e[start_pos+j-1] = tdeg[j]-degs[i][j]
+       end
+       push_term!(F, c, e)
+   end
+
+   return finish(F)
+end
+
+function _homogenization(f::MPolyElem, W::fmpz_mat, var::String, pos::Int = 1)
+  R = parent(f)
+  A = String.(symbols(R))
+  l = length(A)
+  if (pos > l+1) ||  (pos <1)
+      throw(ArgumentError("Index out of range."))
+  end
+  if size(W, 1) == 1
```

Comment:
Personally, I prefer `nrows` and `ncols`, so:
```suggestion
  if nrows(W) == 1
```